src/lib.rs

```diff
@@ -0,0 +1,153 @@
+                block_timestamp: blk.clone()
+                .header
+                .as_ref()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap()
+                .seconds
+                .to_string(),
+                log_ordinal: log.ordinal(),
+                ..Default::default()
+            };
+    
+            air_transfers.push(air_transfer);
+        }
+        if let Some(event) = abis::erc721::events::Transfer::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            // log::info!("ERC721 token: {}, {}", log.receipt.transaction.hash.clone().to_string(), log.ordinal());
+            let air_transfer: AirTransfer = AirTransfer{
+                amount: 1.to_string(),
+                token_id: event.token_id.to_string(),
+                hash: Hex(log.receipt.transaction.hash.clone()).to_string(),
+                from: Hex(event.from).to_string(),
+                to: Hex(event.to).to_string(),
+                token_type:2,
+                block_number: blk.clone().number,
+                block_timestamp: blk.clone()
+                .header
+                .as_ref()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap()
+                .seconds
+                .to_string(),
+                log_ordinal: log.ordinal(),
+                ..Default::default()
+            };
+            air_transfers.push(air_transfer);
+        }
+        if let Some(event) = abis::erc1155::events::TransferBatch::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            // log::info!("transfer values: {}", &event.values);
+            let air_transfer_batch: AirErc1155TransferBatch = AirErc1155TransferBatch{
+                amounts: event.values.iter().map(|c| c.clone().to_string()).collect(),
+                hash: Hex(log.receipt.transaction.hash.clone()).to_string(),
+                from: Hex(event.from).to_string(),
+                to: Hex(event.to).to_string(),
+                token_type:3,
+                block_number: blk.clone().number,
+                block_timestamp: blk.clone()
+                .header
+                .as_ref()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap()
+                .seconds
+                .to_string(),
+                token_ids: event.ids.iter().map(|id| Hex(id.clone()).0.to_string()).collect(),
+                log_ordinal: log.ordinal(),
+                ..Default::default()
+            };
+            air_transfer_batchs.push(air_transfer_batch);
+        }
+        if let Some(event) = abis::erc1155::events::TransferSingle::match_and_decode(log){
+            log::info!("transfer event: {}", Hex(&event.to));
+            let air_transfer_single: AirErc1155TransferSingle = AirErc1155TransferSingle{
```

Comment:
please rename this to erc1155_transfer_single

Reply:
Done